_posts/2017-08-10-0.1-using-network-policy.md

```diff
@@ -0,0 +1,139 @@
+
+While this won’t let an attacker access other Istio-enabled pods, so long as they are correctly configured, it opens several attack vectors:
+
+- Attacking unprotected pods
+- Attempting to deny service to protected pods by sending lots of traffic
+- Exfiltrating data collected in the pod
+- Attacking the cluster infrastructure (servers or Kubernetes services)
+- Attacking services outside the mesh, like databases, storage arrays, or legacy systems.
+
+Network Policy is typically enforced at the host node, outside the network namespace of the guest pods. This means that compromised or misbehaving pods must break into the root namespace to avoid enforcement. With the addition of egress policy due in Kubernetes 1.8, this difference makes Network Policy a key part of protecting your infrastructure from compromised workloads.
+
+## Examples
+
+Let’s walk through a few examples of what you might want to do with Kubernetes Network Policy for an Istio-enabled application.  Consider the BookInfo sample application.  We’re going to cover the following use cases for Network Policy:
+
+- Reduce attack surface of the application ingress
+- Enforce fine-grained isolation within the application
+
+### Reduce attack surface of the application ingress
+
+Our application ingress controller is the main entry-point to our application from the outside world.  A quick peek at istio.yaml (used to install Istio) defines the Istio ingress like this:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-ingress
+  labels:
+    istio: ingress
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 80
+    name: http
+  - port: 443
+    name: https
+  selector:
+    istio: ingress
+```
+
+The istio-ingress exposes ports 80 and 443.  Let’s limit incoming traffic to just these two ports.  Envoy has a [built-in administrative interface](https://lyft.github.io/envoy/docs/operations/admin.html#operations-admin-interface), and we don’t want a misconfigured istio-ingress image to accidentally expose our admin interface to the outside world.  This is an example of defense in depth: a properly configured image should not expose the interface, and a properly configured Network Policy will prevent anyone from connecting to it.  Either can fail or be misconfigured and we are still protected.
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: istio-ingress-lockdown
+  namespace: default
+spec:
+  podSelector:
+    matchLabels:
+      istio: ingress
+  ingress:
+  - ports:
+    - protocol: TCP
+      port: 80
+    - protocol: TCP
+      port: 443
+```
+
+### Enforce fine-grained isolation within the application
+
+Here is the service graph for the BookInfo application.
+
+<figure>
+    <a href="{{home}}/img/bookinfo-withistio.svg">
+    <img style="max-width: 100%;" src="{{home}}/img/bookinfo-withistio.svg" alt="BookInfo service graph" title="BookInfo service graph" />
+    </a>
+</figure>
+
```

Comment:
This PR is almost good to go. Thanks a lot for all the work @spikecurtis . Two small nits: can you remove the images from your PR? We already have the same bookinfo images in this repo. If you could refer to those images, it would reduce the overhead.